ghost/in-memory-repository/src/InMemoryRepository.ts

```diff
@@ -0,0 +1,80 @@
+import nql from '@tryghost/nql';
+
+type Entity<T> = {
+    id: T;
+    deleted: boolean;
```

Comment:
@allouis I'm a confused about this "deleted" property. Why do we need to have it around instead of removing the Entity from the store alltogether?  I'm missing some context probably, so wanted to understand what use case or reasoning led to such data structure.